src/test/scala/in/ashwanthkumar/suuchi/partitioner/ConsistentHashRingSpec.scala

```diff
@@ -0,0 +1,37 @@
+package in.ashwanthkumar.suuchi.partitioner
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers._
+
+object IdentityHash extends Hash {
+  override def hash(bytes: Array[Byte]): Integer = bytes.toString.toInt
+}
+
+class ConsistentHashRingSpec extends FlatSpec {
+  "ConsistentHashRing" should "pin nodes into the ring accounting for virtual nodes" in {
+    val ring = new ConsistentHashRing(SuuchiHash, 3)
```

Comment:
@ashwanthkumar Actually, No. I might have to change things a little bit there to get better tests going. Will chip away at it in parallel